tests/main/views/test_direct_award.py

```diff
@@ -326,3 +327,20 @@ def test_search_api_urls_convert_to_correct_frontend_urls(self, search_api_url,
         doc = html.fromstring(body)
         assert html_escape(frontend_url) in body
         assert len(doc.xpath('//a[@href="{}"]'.format(frontend_url)))
+
+
+class TestDirectAwardEndSearch(TestDirectAwardBase):
+    def test_end_search_page_renders(self):
+        self.login_as_buyer()
+        res = self.client.get('/buyers/direct-award/g-cloud/projects/1/end-search')
+        assert res.status_code == 200
+        doc = html.fromstring(res.get_data(as_text=True))
+        assert len(doc.xpath('//h1[contains(normalize-space(), "End search")]')) == 1
+
+    def test_end_search_redirects_to_project_page(self):
+        self.login_as_buyer()
+        data_api_client.lock_direct_award_project = mock.Mock()
+        data_api_client.lock_direct_award_project.return_value = self._get_direct_award_lock_project_fixture()
+        res = self.client.post('/buyers/direct-award/g-cloud/projects/1/end-search')
+        assert res.status_code == 302
+        assert re.findall(r'\/buyers\/direct-award\/g-cloud\/projects\/1$', res.location)
```

Comment:
Can you just use the `in` operator here, if so it'd be a lot more readable than the regex.

Reply:
Or just res.location.endswith.

And if you're using raw strings you shouldn't need to escape the slashes.